ParallelPrimitives/RadixSort.cpp

```diff
@@ -111,194 +114,255 @@ void RadixSort::compileKernels( const std::string& kernelPath, const std::string
+//		}
+//
+//#endif
+//		if( m_flags == Flag::LOG )
+//		{
+//			printKernelInfo( record.kernelName, oroFunctions[record.kernelType] );
+//		}
+//	}
+
+	// TODO: bit code support?
+#define LOAD_FUNC( var, kernel ) var = m_oroutils.getFunctionFromFile( m_device, currentKernelPath.c_str(), kernel, &opts );
+	LOAD_FUNC( m_gHistogram, "gHistogram" );
+	LOAD_FUNC( m_gPrefixSum, "gPrefixSum" );
+	LOAD_FUNC( m_onesweep_reorderKey64, "onesweep_reorderKey64" );
+	LOAD_FUNC( m_onesweep_reorderKeyPair64, "onesweep_reorderKeyPair64" );
+#undef LOAD_FUNC
 }
 
-int RadixSort::calculateWGsToExecute( const int blockSize ) const noexcept
-{
-	const int warpPerWG = blockSize / m_warp_size;
-	const int warpPerWGP = m_props.maxThreadsPerMultiProcessor / m_warp_size;
-	const int occupancyFromWarp = ( warpPerWGP > 0 ) ? ( warpPerWGP / warpPerWG ) : 1;
-
-	const int occupancy = std::max( 1, occupancyFromWarp );
-
-	if( m_flags == Flag::LOG )
-	{
-		std::cout << "Occupancy: " << occupancy << '\n';
-	}
-
-	static constexpr auto min_num_blocks = 16;
-	auto number_of_blocks = m_props.multiProcessorCount > 0 ? m_props.multiProcessorCount * occupancy : min_num_blocks;
-
-	if( m_num_threads_per_block_for_scan > BIN_SIZE )
-	{
-		// Note: both are divisible by 2
-		const auto base = m_num_threads_per_block_for_scan / BIN_SIZE;
-
-		// Floor
-		number_of_blocks = ( number_of_blocks / base ) * base;
-	}
-
-	return number_of_blocks;
-}
+//int RadixSort::calculateWGsToExecute( const int blockSize ) const noexcept
+//{
+//	const int warpSize = ( m_props.warpSize != 0 ) ? m_props.warpSize : DEFAULT_WARP_SIZE;
+//	const int warpPerWG = blockSize / warpSize;
+//	const int warpPerWGP = m_props.maxThreadsPerMultiProcessor / warpSize;
+//	const int occupancyFromWarp = ( warpPerWGP > 0 ) ? ( warpPerWGP / warpPerWG ) : 1;
+//
+//	const int occupancy = std::max( 1, occupancyFromWarp );
+//
+//	if( m_flags == Flag::LOG )
+//	{
+//		std::cout << "Occupancy: " << occupancy << '\n';
+//	}
+//
+//	static constexpr auto min_num_blocks = 16;
+//	auto number_of_blocks = m_props.multiProcessorCount > 0 ? m_props.multiProcessorCount * occupancy : min_num_blocks;
+//
+//	if( m_num_threads_per_block_for_scan > BIN_SIZE )
+//	{
+//		// Note: both are divisible by 2
+//		const auto base = m_num_threads_per_block_for_scan / BIN_SIZE;
+//
+//		// Floor
+//		number_of_blocks = ( number_of_blocks / base ) * base;
+//	}
+//
+//	return number_of_blocks;
+//}
 
 void RadixSort::configure( const std::string& kernelPath, const std::string& includeDir ) noexcept
 {
 	compileKernels( kernelPath, includeDir );
 
-	m_num_blocks_for_count = calculateWGsToExecute( m_num_threads_per_block_for_count );
+	u64 gpSumBuffer = sizeof( u32 ) * 256 * sizeof( u32 /* key type */ );
+	u64 lookBackBuffer = next_multiple64( sizeof( u64 ) * ( 256 * LOOKBACK_TABLE_SIZE + 1 ), 16 );
+	m_tmpBuffer.resize( gpSumBuffer + lookBackBuffer );
```

Comment:
We need to use `resizeAsync` with the custom stream here: https://github.com/amdadvtech/Orochi/blob/main/ParallelPrimitives/RadixSort.cpp#L257

Reply:
I changed